### PR TITLE
Refactor source compiler

### PR DESCRIPTION
**Pull request by thobe**
Allow for simple source code rewriting.

Move source code templates to the same place.

### CODE REVIEW

**Comment by thobe** [comment]
Re-opening #4962 with a better message about what it actually contains.

File: community/codegen/src/main/java/org/neo4j/codegen/source/ClassWriter.java [hunk 1]
```diff
@@ -33,10 +33,51 @@
...
         this.target = target;
+        this.configuration = configuration;
+    }
+
+    void declarePackage( TypeReference type )
+    {
+        append( "package " ).append( type.packageName() ).append( ";\n" );
+    }
+
+    void javadoc( String javadoc )
```
Comment (systay) [inline community/codegen/src/main/java/org/neo4j/codegen/source/ClassWriter.java:49]: When do you expect us to want this for our compiled plans?

Comment (thobe) [inline community/codegen/src/main/java/org/neo4j/codegen/source/ClassWriter.java:49]: We already use it, the generated source code contains a comment saying "this is generates code".

We had that before this refactoring, this refactoring just makes it more clear what it is.

Comment (thobe) [inline community/codegen/src/main/java/org/neo4j/codegen/source/ClassWriter.java:49]: But perhaps we should use the `javax.annotation.Generated` annotation instead.

**Pull request merged**